### DISCUSSION TITLE
New Command - Options - Layouts - View toggles

### DISCUSSION THREAD

**Discussion by advice2020** [doublecmd/doublecmd#1543] [Suggestions]
The `cm_Show` based commands are used to toggle show/hide items listed in `Options - Layout` section.
Suggesting to add other items that do not currently have commands for them
-Middle Toolbar
-Drive Buttons
-Drives List Button
-Folder Tabs
-Current Directory (Address Bar)
-Tabstop Header (Column Header)
-Status Bar
-Command Line
-Function Key Buttons
-Log Window
-Terminal Window
I personally do not need all of them but am just mentioning all the ones that do not have commands for them.
Tough decision because lot of commands, could just be added as parameters to a single command (just then would not get specific icons for each).

This idea would obviously allow user to be able to turn specific Layout items on / off but could also allow user the ability to string a whole bunch of DC commands together (using Lua) to almost be able to build a “Change Layout” command. 
Meaning this could allow users to be able to create specific DC “Layouts” for specific tasks and swap between them without having to create separate settings folders for DC instances.

Thank You

**Comment by Skif-off** [comment]
[cm_Options](https://doublecmd.github.io/doc/en/cmds.html#cm_Options) has parameters:
command: `cm_Options`
params: `TfrmOptionsLayout`

P.S. The plan was to add access to some DC settings from Lua scripts.

**Comment by advice2020** [comment]
@Skif-off
I apologize for tagging you but I deleted my last reply for this post and replaced it with this one.  
I thought I was able to get this to work in the past but now I cannot get this to work.

MAIN TOOLBAR
In `Options - Layout` there is `Show Toolbar` toggle to turn Main Toolbar on/off.
DC offers an actual command for this `cm_ShowButtonMenu`

MIDDLE TOOLBAR
Could you or anyone else reading this, provide an example for say toggling the visibility of the Middle Toolbar on/off?
I feel that this should be a parameter within the `cm_ShowButtonMenu` command.
Is there a way to achieve this using `cm_Options` parameters you mentioned?
Or do these parameters only open inthe the `Options` window itself?

TAB BAR
And or the toggle to turn Tab bar on/off

Thank You

**Comment by advice2020** [comment]
Still interested if anyone is willing to confirm this.
Thank You
